google-cloud-clients/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Model.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.cloud.bigquery.BigQuery.ModelOption;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A Google BigQuery ML Model.
+ *
+ * <p>Objects of this class are immutable. Operations that modify the table like {@link #update}
+ * return a new object. To get a {@code Model} object with the most recent information use {@link
+ * #reload}.
+ */
+public class Model extends ModelInfo {
+
+  private final BigQueryOptions options;
+  private transient BigQuery bigquery;
+
+  public static class Builder extends ModelInfo.Builder {
+
+    private final BigQuery bigquery;
+    private final ModelInfo.BuilderImpl infoBuilder;
+
+    Builder(BigQuery bigquery, ModelId modelId) {
+      this.bigquery = bigquery;
+      this.infoBuilder = new ModelInfo.BuilderImpl();
+      this.infoBuilder.setModelId(modelId);
+    }
+
+    Builder(Model model) {
+      this.bigquery = model.bigquery;
+      this.infoBuilder = new ModelInfo.BuilderImpl(model);
+    }
+
+    @Override
+    Builder setCreationTime(Long creationTime) {
```

Comment:
Should these be Java / Joda date-times? (I guess we haven't dropped Java 7 yet, so maybe not...)

Reply:
Consistency-wise, this mirrors the existing usages.  It looks like we're still announcing a java 7 compatibility for google-cloud-java.